catalog/db_branch_scanner.go

```diff
@@ -35,9 +42,37 @@ func NewDBBranchScanner(tx db.Tx, branchID int64, commitID CommitID, opts *DBSca
 		s.opts.BufferSize = DBScannerDefaultBufferSize
 	}
 	s.buf = make([]*DBScannerEntry, 0, s.opts.BufferSize)
+	commitsWhere, _ := getRelevantCommitsCondition(tx, branchID, commitID)
```

Comment:
1. error ignored
2. can be called once in `readBufferIfNeeded` or part of query build and not on iterator creation